modules/ldapbackend/dnssec.cc

```diff
@@ -0,0 +1,1416 @@
+/*
+ * This file is part of PowerDNS or dnsdist.
+ * Copyright -- PowerDNS.COM B.V. and its contributors
+ * originally authored by Norbert Sendetzky
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * In addition, for the avoidance of any doubt, permission is granted to
+ * link this program with OpenSSL and to (re)distribute the binaries
+ * produced as the result of such linking.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#include "exceptions.hh"
+#include "ldapbackend.hh"
+#include <cstdlib>
+
+
+bool LdapBackend::doesDNSSEC()
+{
+  return m_dnssec;
+}
+
+
+std::string LdapBackend::getDomainMetadataDN( const DNSName &name )
+{
+  std::string dn;
+  std::string filter = strbind( ":domain:", m_pldap->escape( name.toStringRootDot() ), getArg( "metadata-searchfilter" ) );
+  const char* attributes[] = { "objectClass", NULL };
+  PowerLDAP::sentry_t result;
+
+  try {
+    PowerLDAP::SearchResult* search = m_pldap->search( m_metadata_searchdn, LDAP_SCOPE_SUBTREE, filter, attributes );
+
+    if ( search->getNext( result, true ) ) {
```

Comment:
It looks like `PowerLDAP::SearchResult::getNext()` can throw, which would result in `search` being lost. How about making `PowerLDAP::search()` return a `unique_ptr` instead?